cmd/sniproxy/snipproxy.go

```diff
@@ -107,11 +133,88 @@ func (s *server) serveConn(c net.Conn) {
 		return netutil.NewOneConnListener(c, nil), nil
 	}
 	p.AddSNIRouteFunc(addrPortStr, func(ctx context.Context, sniName string) (t tcpproxy.Target, ok bool) {
-		log.Printf("got req for %q from %v", sniName, c.RemoteAddr())
 		return &tcpproxy.DialProxy{
 			Addr:        net.JoinHostPort(sniName, port),
 			DialContext: dialer.DialContext,
 		}, true
 	})
 	p.Start()
 }
+
+// getAddresses returns the tsnet IP addresses of this process
+func (s *server) getAddresses() (ip4, ip6 netip.Addr) {
```

Comment:
we can probably cache these, they won't change during the duration of the process.

Reply:
The case where the IPs do change is the very first time the process starts, if authentication doesn't happen right away. Either they don't supply an authkey and they eventually get around to visiting the LoginURL, or the authkey takes some time to reach the coordination server.

Added caching but only if the addresses are valid.

Reply:
Removed getAddresses, now calls `s.ts.TailscaleIPs()`